aptos-move/aptos-e2e-comparison-testing/src/online_execution.rs

```diff
@@ -0,0 +1,240 @@
+        let mut module_registry_map = HashMap::new();
+        while cur_version < begin + limit {
+            let batch = if cur_version + self.batch_size <= begin + limit {
+                self.batch_size
+            } else {
+                begin + limit - cur_version
+            };
+            let res_txns = self
+                .debugger
+                .get_and_filter_committed_transactions(
+                    cur_version,
+                    batch,
+                    self.filter_condition,
+                    &mut module_registry_map,
+                )
+                .await;
+            // if error happens when collecting txns, log the version range
+            if res_txns.is_err() {
+                index_writer.lock().unwrap().write_err(&format!(
+                    "{}:{}:{:?}",
+                    cur_version,
+                    batch,
+                    res_txns.unwrap_err()
+                ));
+                cur_version += batch;
+                continue;
+            }
+            let txns = res_txns.unwrap_or_default();
+            if !txns.is_empty() {
+                let mut txn_execution_ths = vec![];
+                for (version, txn, source_code_data) in txns {
+                    println!("get txn at version:{}", version);
+
+                    let compilation_cache = compilation_cache.clone();
+                    let current_dir = self.current_dir.clone();
+                    let execution_mode = self.execution_mode;
+                    let endpoint = self.endpoint.clone();
+
+                    let txn_execution_thread = tokio::task::spawn_blocking(move || {
+                        let executor = crate::Execution::new(current_dir.clone(), execution_mode);
+
+                        let mut version_idx = TxnIndex {
+                            version,
+                            txn: txn.clone(),
+                            package_info: PackageInfo::non_compilable_info(),
+                        };
+
+                        // handle source code
+                        if let Some((address, package_name, map)) = source_code_data {
+                            let execution_mode_opt = Some(execution_mode);
+                            let package_info_opt = Self::dump_and_check_src(
+                                version,
+                                address,
+                                package_name,
+                                map,
+                                &mut compilation_cache.lock().unwrap(),
+                                execution_mode_opt,
+                                current_dir.clone(),
+                            );
+                            if package_info_opt.is_none() {
+                                return;
+                            }
+
+                            version_idx.package_info = package_info_opt.unwrap();
+
+                            let state_store = FakeDataStore::default();
```

Comment:
This is just an empty hashmap inside? Why do we need it?

Reply:
state_store refers to the data state (code excluded), which is only used in the offline execution mode when data state is stored in the file. In the online execution mode, we use DebuggerStateView for the data state so it is empty. We can discuss more on how to refactor it